client/lib/sites-list/list.js

```diff
@@ -404,7 +404,11 @@ SitesList.prototype.getSite = function( siteID ) {
 		// clashes between a domain redirect and a Jetpack site, as well as domains
 		// on subfolders, but we also need to look for the `domain` as a last resort
 		// to cover mapped domains for regular WP.com sites.
-		return site.ID === siteID || site.slug === siteID || site.domain === siteID || site.wpcom_url === siteID;
+		return site.ID === siteID ||
+			site.slug === siteID ||
+			site.domain === siteID ||
+			site.wpcom_url === siteID ||
+			( site.options.is_domain_only && site.name === siteID );
```

Comment:
Why isn't `site.slug` set after checkout? Is there some sort of delay in the API?